astropy/units/decorators.py

```diff
@@ -250,12 +308,24 @@ def wrapper(*func_args, **func_kwargs):
             with add_enabled_equivalencies(self.equivalencies):
                 return_ = wrapped_function(*func_args, **func_kwargs)
 
-            valid_empty = (inspect.Signature.empty, None)
-            if wrapped_signature.return_annotation not in valid_empty:
-                return return_.to(wrapped_signature.return_annotation)
-            else:
+            # Return
+            # if there is no annotation, return as-is
+            valid_empty = (inspect.Signature.empty, None, NoneType)
+            if wrapped_signature.return_annotation in valid_empty:
                 return return_
 
+            # annotated output
+            ret_targets = _parse_annotation(
+                wrapped_signature.return_annotation)
+            # TODO for future PR: drop support for Quantity when mean Unit
+            # TODO for future PR: allow for generic annotations.
+            #   # This allows for non unit/Quantity/str annotations.
```

Comment:
Is this still relevant? It does not look one could use these lines of code, so just remove them?

Reply:
Fixed. I merged it into the previous lines.